### ISSUE TITLE
Return type ignore

### ISSUE THREAD

**Issue by rzwq** [astral-sh/ty#4146]
### Question

I little bit confused, while i writing this example:

```python
from functools import wraps
from typing import Callable


class Human: ...


class Soprano:
    def __init__(self):
        self.family = set()

    def __call__(self, func: Callable[..., Human]) -> Callable[..., Human]:

        @wraps(func)
        def wrapper(*args, **kwargs):
            human = func(*args, **kwargs)
            self.family.add(human)
            print(f"{human} now in Soprano's family!")
            return human

        return wrapper


familia = Soprano()


@familia
def Tony():
    return


if __name__ == "__main__":
    Tony()
```

While **mypy** and **ty** got *false positive*, **pyright** find potential bug in `Tony`.

### Version

0.0.65

**Comment by sharkdp** [comment]
ty doesn't have return type inference, so the return type is inferred as `Unknown`, which makes the `familia` call pass. See https://github.com/astral-sh/ty/issues/128 for details.
